packages/framework/tests/Unit/Component/Translation/ConstraintViolationExtractorTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Component\Translation;
+
+use JMS\TranslationBundle\Model\FileSource;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Model\MessageCatalogue;
+use PhpParser\ParserFactory;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Translation\ConstraintViolationExtractor;
+use SplFileInfo;
+
+class ConstraintViolationExtractorTest extends TestCase
+{
+    public function testMessagesAreExtractedFromConstraintViolationClass(): void
+    {
+        $file = new SplFileInfo(__DIR__ . '/Resources/ConstraintViolationClass.php');
+
+        $actualCatalogue = $this->extract($file);
+
+        $expectedCatalogue = new MessageCatalogue();
+
+        $message = new Message('This message will be extracted into "validators" translation domain', 'validators');
```

Comment:
Maybe it would make sense to put the sentence into some constant so it is obvious that it is the same message as the one defined `ConstraintViolationClass`. However, I think it will not work then, right?

Reply:
Yeah, It's mentioned in `ConstraintViolationClass` annotation. Violations cannot be variables (and constants).